### PR TITLE
[CN-1540] refactored logger and metrics into their own files

### PR DESCRIPTION
**Pull request by Elliot-Ball-Tray**
refactored logger and metrics into their own files these functions are now bound to falafel

### CODE REVIEW

File: lib/logger/bindLogger.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const { initLogger } = require('./logger');
+const { initDataDogMetricsApi } = require('./metricsApiWrapper');
+
+const initDependencies = async () => {
+	const logVersion = 'v1';
+	if (process.env.FALAFEL_ENABLE_LOGGING !== 'true') {
+		return { metricsApiWrapper: undefined, logger: undefined };
+	}
+	try {
+		const apiKey = await getDatadogSecret(process.env.DATADOG_SSM_PARAM_KEY, process.env.AWS_REGION);
```
Comment (gnyani-trayio) [inline lib/logger/bindLogger.js:12]: personally I would prefer not to pass ENVs as variables unless there are cases where we would have to invoke the function with different ENVs.

File: lib/logger/bindLogger.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		const metricsApiWrapper = await initDataDogMetricsApi(process.env.DATADOG_HOST, apiKey, logVersion);
+		const logger = initLogger(process.env.DATADOG_HOST, apiKey, logVersion);
+		return { metricsApiWrapper, logger };
+	} catch (error) {
+		if (process.env.FALAFEL_ENABLE_LOGGING === 'true') {
```
Comment (gnyani-trayio) [inline lib/logger/bindLogger.js:17]: this if statement is redundant, as the execution flow would not come here if logging was not enabled.

File: lib/logger/bindLogger.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+const initDependencies = async () => {
+	const logVersion = 'v1';
+	if (process.env.FALAFEL_ENABLE_LOGGING !== 'true') {
+		return { metricsApiWrapper: undefined, logger: undefined };
+	}
+	try {
+		const apiKey = await getDatadogSecret(process.env.DATADOG_SSM_PARAM_KEY, process.env.AWS_REGION);
+		const metricsApiWrapper = await initDataDogMetricsApi(process.env.DATADOG_HOST, apiKey, logVersion);
+		const logger = initLogger(process.env.DATADOG_HOST, apiKey, logVersion);
```
Comment (gnyani-trayio) [inline lib/logger/bindLogger.js:14]: I would probably not pass DATADOG_HOST as a param here.

File: lib/index.js [hunk 4]
```diff
@@ -86,6 +87,7 @@ var Falafel = function () {
 
 			}
 
+			const dependenciesLoaded = bindLogger();
```
Comment (gnyani-trayio) [inline lib/index.js:90]: wouldn't we need to await?

**Review by gnyani-trayio — DISMISSED** [review]

**Review by gnyani-trayio — APPROVED** [review]

**Review by johnbastian-trayio — APPROVED** [review]
LGTM

**Review by gnyani-trayio — APPROVED** [review]

**Review by johnbastian-trayio — APPROVED** [review]

**Pull request merged**